test/e2e/service.go

```diff
@@ -34,11 +34,15 @@ import (
 
 var _ = Describe("Services", func() {
 	var c *client.Client
+	// Use these in tests.  They're unique for each test to prevent name collisions.
+	var namespace0, namespace1 string
 
 	BeforeEach(func() {
 		var err error
 		c, err = loadClient()
 		Expect(err).NotTo(HaveOccurred())
+		namespace0 = "e2e-ns-" + dateStamp() + "-0"
```

Comment:
PS: I suggest merging as is for now, and adding namespace deletion in a second pass.  What we have here already is a vast improvement over the status quo.  Adding namespace deletion will be a further improvement.